File: app/src/main/java/com/xephorium/crystalnote/ui/selection/SelectionPresenter.java
package com.xephorium.crystalnote.ui.selection;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

import com.xephorium.crystalnote.data.NoteRepository;
import com.xephorium.crystalnote.data.SharedPreferencesRepository;
import com.xephorium.crystalnote.data.util.NoteUtils;
import com.xephorium.crystalnote.data.util.NoteUtils.SortType;
import com.xephorium.crystalnote.data.model.Note;
import com.xephorium.crystalnote.ui.IntentLibrary;
import com.xephorium.crystalnote.ui.base.BasePresenter;
import com.xephorium.crystalnote.ui.creation.CreationActivity;

public class SelectionPresenter extends BasePresenter<SelectionView> {

    private Context context;
    private NoteRepository noteRepository;

    public SelectionPresenter(SelectionView view) {
        attachView(view);
        this.context = (Context) view;
        this.noteRepository = new NoteRepository(context);
    }

    public void refreshNoteList() {
        getView().populateNoteList(NoteUtils.sortNotes(noteRepository.getNotes(), SortType.DATE_NEW));
    }

    public void handleToolbarBackClick() {
        getView().closeSelectionActivity();
    }

    public void handleActionButtonClick() {
        Intent intent = new Intent(context, CreationActivity.class);
        intent.setAction(IntentLibrary.CREATE_NOTE);
        ((Activity) getView()).startActivity(intent);
    }

    public void handleNoteClick(Note note) {
        SharedPreferencesRepository.setDisplayNoteName(context, note);
        refreshWidget();
        getView().closeSelectionActivity();
    }

    public boolean handleNoteLongClick(Note note) {
        // Do Nothing
        return true;
    }

    private void refreshWidget() {
        Intent intent = new Intent();
        intent.setAction(IntentLibrary.UPDATE_NOTE_INTENT);
        context.sendBroadcast(intent);
    }
}

// Text Edit Launch Intent
// Intent intent = new Intent(Intent.ACTION_VIEW);
// Uri uri = Uri.parse("file://" + note.getPath());
// intent.setDataAndType(uri, "text/plain");
// this.startActivity(intent);

File: app/src/main/java/com/xephorium/crystalnote/ui/selection/SelectionPresenter.kt
package com.xephorium.crystalnote.ui.selection

import com.xephorium.crystalnote.data.utility.NoteUtility
import com.xephorium.crystalnote.data.utility.NoteUtility.SortType
import com.xephorium.crystalnote.data.model.Note

class SelectionPresenter : SelectionContract.Presenter() {


    /*--- Lifecycle Methods ---*/

    override fun attachView(view: SelectionContract.View) {
        super.attachView(view)

        refreshNoteList()
    }


    /*--- Action Handling Methods ---*/

    override fun handleToolbarBackClick() {
        view?.navigateBack()
    }

    override fun handleNewNoteButtonClick() {
        view?.navigateToNewNote()
    }

    override fun handleNoteClick(note: Note) {
        sharedPreferencesRepository.setDisplayNoteName(note)
        view?.refreshWidget()
        view?.navigateBack()
    }

    override fun handleNoteLongClick(note: Note) {
        // Do Nothing
    }

    override fun handleNoteListRefresh() {
        refreshNoteList()
    }


    /*--- Private Methods ---*/

    private fun refreshNoteList() {
        view?.populateNoteList(NoteUtility.sortNotes(
                noteRepository.getNotes().toMutableList(),
                SortType.DATE_NEW
        ))
    }
}

// Text Edit Launch Intent
// Intent intent = new Intent(Intent.ACTION_VIEW);
// Uri uri = Uri.parse("file://" + note.getPath());
// intent.setDataAndType(uri, "text/plain");
// this.startActivity(intent);